### PR TITLE
DL-5279 Auth Client Upgrade

### PR DESCRIPTION
**Pull request by MarjanMubarok**
# JIRA Story or Github Number - descriptive title

**Bug fix / New feature** (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## Checklist Reviewee

 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [ ]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date
 
 ## Checklist Reviewer (Name here)
 
  - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
  - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
  - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions
  - [ ]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
  - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientmandate/connectors/EtmpConnector.scala [hunk 1]
```diff
@@ -119,9 +126,9 @@ trait EtmpConnector extends RawResponseReads with Auditable {
...
   }
 
-  private def createHeaderCarrier: HeaderCarrier = {
-    HeaderCarrier(extraHeaders = Seq("Environment" -> urlHeaderEnvironment),
-      authorization = Some(Authorization(urlHeaderAuthorization)))
-  }
+//  private def createHeaderCarrier: HeaderCarrier = {
+//    HeaderCarrier(extraHeaders = Seq("Environment" -> urlHeaderEnvironment),
+//      authorization = Some(Authorization(urlHeaderAuthorization)))
+//  }
```
Comment (paulanderson1234) [inline app/uk/gov/hmrc/agentclientmandate/connectors/EtmpConnector.scala:132]: So are these needed or not ?

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -12,7 +12,7 @@ object AppDependencies {
     "com.typesafe.akka" %% "akka-stream" % "2.6.14",
     "com.typesafe.akka" %% "akka-slf4j" % "2.6.14",
     ws,
-    "uk.gov.hmrc"%% "bootstrap-backend-play-27"  % "3.4.0",
+    "uk.gov.hmrc"%% "bootstrap-backend-play-27"  % "4.1.0",
```
Comment (paulanderson1234) [inline project/AppDependencies.scala:15]: I tjought we were upgrading to 5.2.0 ?

**Comment by paulanderson1234** [comment]
There are 16 warnings coming up on compilation when you run with : scalacOptions ++= Seq("-feature") in build.sbt

postfix operator seconds should be enabled
by making the implicit value scala.language.postfixOps visible.
This can be achieved by adding the import clause 'import scala.language.postfixOps'

File: app/uk/gov/hmrc/agentclientmandate/connectors/EtmpConnector.scala (hunk 1 above)
Comment (MarjanMubarok) [reply app/uk/gov/hmrc/agentclientmandate/connectors/EtmpConnector.scala:132]: Done

**Review by paulanderson1234 — APPROVED** [review]

**Pull request merged**